components/script/script_runtime.rs

```diff
@@ -776,3 +780,44 @@ impl Deref for JSContext {
         &self.0
     }
 }
+
+pub struct StreamConsumer(*mut JSStreamConsumer);
+
+#[allow(unsafe_code)]
+impl StreamConsumer {
+    fn consumeChunk(sc: &mut JSStreamConsumer, begin: *const u8, length: usize) -> bool {
+        unsafe {
+            return StreamConsumerConsumeChunk(sc, begin, length);
+        }
+    }
+
+    fn streamEnd(sc: &mut JSStreamConsumer){
+        unsafe {
+            StreamConsumerStreamEnd(sc);
+        }
+    }
+
+    fn streamError(sc: &mut JSStreamConsumer, errorCode: usize){
+        unsafe {
+            StreamConsumerStreamError(sc, errorCode);
+        }
+    }
+
+    fn noteResponseURLs(
+        sc: &mut JSStreamConsumer,
+        maybeUrl: *const c_char,
+        maybeSourceMapUrl: *const c_char
+    ) {
+        unsafe {
+            StreamConsumerNoteResponseURLs(sc, maybeUrl, maybeSourceMapUrl);
+        }
+    }
+}
+
+#[allow(unsafe_code)]
+unsafe extern "C" fn ReportStreamErrorCallback(cx: *mut JSContext, errorCode: usize){
+    error!(
+        "Error initializing StreamConsumer: {:?}",
+        RUST_js_GetErrorMessage(cx as *mut c_void, errorCode as u32)
```

Comment:
The first argument to this API isn't actually used (https://searchfox.org/mozilla-central/rev/74cc0f4dce444fe0757e2a6b8307d19e4d0e0212/js/src/vm/JSContext.cpp#995) so let's pass `ptr::null_mut()` instead.

Reply:
Done